client/ayon_blender/plugins/load/load_cache.py

```diff
@@ -69,31 +69,22 @@ def _process(self, libpath, asset_group, group_name):
                 relative_path=relative
             )
 
-        imported = lib.get_selection()
-
-        # Use first EMPTY without parent as container
-        container = next(
-            (obj for obj in imported
-             if obj.type == "EMPTY" and not obj.parent),
-            None
-        )
-
         objects = []
-        if container:
-            nodes = list(container.children)
+        uninspected_top_objs = [obj for obj in lib.get_selection() if not obj.parent]
 
-            for obj in nodes:
-                obj.parent = asset_group
+        while uninspected_top_objs:
```

Comment:
Is there any reason why we use while loop instead of for loop, knowing that we can iterate through `uninsepcted_top_objs` ?
We can do something like this:
```python
        uninspected_top_objs = [obj for obj in lib.get_selection() if not obj.parent]

        for obj in uninspected_top_objs:
            # Ignore EMPTYs without transforms/parent (blank containers)
            if obj.type == "EMPTY" and obj.matrix_local.is_identity:
                uninspected_top_objs.extend(obj.children)
                bpy.data.objects.remove(obj)
            # Reparent top object to asset_group
            else:
                obj.parent = asset_group
                objects.append(obj)
                objects.extend(obj.children_recursive)
```

Reply:
You're totally right, I guess I confused myself with the `.pop` thingie and modifying the list while iterating.
Simply re-parenting as it comes makes the code even simpler.